fairlearn/_input_validation.py

```diff
@@ -48,6 +48,16 @@ def _validate_and_reformat_input(X, y=None, expect_y=True, enforce_binary_sensit
         values in the `y` data; default False
     :type enforce_binary_labels: bool
     """
+    # scikit-learn's check_array or check_X_y lose the column types. Some estimators rely on this
```

Comment:
See #312. The bottom line: I don't think we need this--hopefully we can just rely on sklearn`s validation / canonicalization routines in our code. Our wrappers should never transform `X`--but some of them may need to get canonical versions of `y` and `sensitive_features`.

Reply:
I've adjusted this accordingly. Let me know what you think @MiroDudik